tox.ini

```diff
@@ -1,39 +1,38 @@
 [tox]
 envlist =
-    py27,py35,py36,py37,pypy,pypy3,coverage,docs
+    py27,py35,py36,py37,py38,pypy,pypy3,coverage,docs
 # Note: if you add new Python versions, please add them to
 # [testenv:coverage] depends as well!
 
 [testenv]
 usedevelop = true
 commands =
-    coverage run -m zope.testrunner --test-path=src []
+    coverage run -p -m zope.testrunner --test-path=src []
+extras =
+    test
 deps =
-    .[test]
     coverage
-setenv =
-   COVERAGE_FILE=.coverage.{envname}
+# Not yet, pending https://github.com/zopefoundation/zope.publisher/issues/49
+# setenv =
+#     ZOPE_INTERFACE_STRICT_IRO=1
 
 [testenv:coverage]
-setenv =
-  COVERAGE_FILE=.coverage
 skip_install = true
 commands =
     coverage erase
     coverage combine
-    coverage html
-    coverage xml
+    coverage html -i
+    coverage xml -i
```

Comment:
`-i` means ignore errors while reading source files.  Why is it needed here?

Reply:
(I see [testenv] has usedevelop = true, which should allow coverage to find the source files without the [`[paths] trick`](https://github.com/zopefoundation/zope.annotation/blob/503f3fc0fede04c97a02576a013401751a43d651/.coveragerc#L8-L12) in .coveragerc.  And .coveragerc has the source = zope.traversing line, which means coverage shouldn't be trying to read the source files of any other package.)

Reply:
Editors that leave temporary files on the disk sometimes make those temporary files look like .py files by name but they are not readable as such by coverage. Emacs does this, so if you have made unsaved changes in some file, without `-i` coverage will fail (it depends on exactly what file and how much in unsaved.)

Reply:
Why would I have unsaved changes? I'll often start a test run while I go back and finish up writing comments or documentation.

Reply:
That is a good reason for having this (but even better would be to file a bug to ask coverage to please not try to read such files!  IIRC emacs's filenames are quite distinct, and not valid as Python module names, with those #'es etc.)

Reply:
coverage claims to have been avoiding those [since 3.2.5](https://bitbucket.org/ned/coveragepy/commits/73afbf09e1fe), but I'm sure I've seen it much more recently than that. I'll have to work up a test case. In the meantime I'll go ahead with these.